Source/WebCore/platform/graphics/transforms/TransformListSharedPrimitivesPrefix.cpp

```diff
@@ -24,37 +24,36 @@
  */
 
 #include "config.h"
-#include "TransformOperationsSharedPrimitivesPrefix.h"
-
-#include "TransformOperations.h"
-#include <algorithm>
+#include "TransformListSharedPrimitivesPrefix.h"
 
 namespace WebCore {
 
-void TransformOperationsSharedPrimitivesPrefix::update(const TransformOperations& operations)
+std::optional<TransformFunctionType> TransformListSharedPrimitivesPrefix::sharedPrimitive(TransformFunctionType a, TransformFunctionType b)
 {
-    size_t maxIteration = operations.size();
-    if (m_indexOfFirstMismatch.has_value())
-        maxIteration = std::min(*m_indexOfFirstMismatch, maxIteration);
-
-    for (size_t i = 0; i < maxIteration; ++i) {
-        Ref operation = operations[i];
-
-        // If we haven't seen an operation at this index before, we can simply use our primitive type.
-        if (i >= m_primitives.size()) {
-            ASSERT(i == m_primitives.size());
-            m_primitives.append(operation->primitiveType());
-            continue;
-        }
-
-        if (auto sharedPrimitive = operation->sharedPrimitiveType(m_primitives[i]))
-            m_primitives[i] = *sharedPrimitive;
-        else {
-            m_indexOfFirstMismatch = i;
-            m_primitives.shrink(i);
-            return;
-        }
+    // https://drafts.csswg.org/css-transforms-2/#interpolation-of-transform-functions
+    // "If both transform functions share a primitive in the two-dimensional space, both transform
+    // functions get converted to the two-dimensional primitive. If one or both transform functions
+    // are three-dimensional transform functions, the common three-dimensional primitive is used."
+
+    if (a == b)
+        return a;
+
+    struct Set {
+        TransformFunctionType transform2D;
+        TransformFunctionType transform3D;
+    };
+    static constexpr std::array<Set, 3> sharedPrimitives = {
```

Comment:
While looking for how we were using `std::array` in this file, I noticed that I think we can omit `<Set, 3>` here.